packages/gestalt/src/useReducedMotion.js

```diff
@@ -0,0 +1,27 @@
+// @flow strict
+import { useState, useEffect } from 'react';
+
+export default function useReducedMotion(): boolean {
+  const supportsMatchMedia = typeof window !== 'undefined' && window.matchMedia;
+
+  const [matches, setMatch] = useState(
+    supportsMatchMedia
+      ? window.matchMedia('(prefers-reduced-motion: reduce)').matches
+      : false
+  );
+  useEffect(() => {
+    if (!supportsMatchMedia) {
+      return () => {};
+    }
+    const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)');
+    const handleChange = () => {
+      setMatch(mediaQuery.matches);
+    };
+    handleChange();
+    mediaQuery.addEventListener('change', handleChange);
+    return () => {
+      mediaQuery.removeEventListener('change', handleChange);
+    };
+  }, [supportsMatchMedia]);
+  return matches;
+}
```

Comment:
Feels like we should have a test for this though I guess it'd just be testing the browser API so maybe not

Reply:
@jennyscript good callout - added tests.

@jennyscript @AlbertCarreras fyi this PR now also ads `@testing-library/react-hooks` to start testing hooks we export.